### PR TITLE
docs(migration): add components migration docs

### LINKED ISSUE #9171: Write initial migration docs for `carbon-components` to `@carbon/styles`
This issue is for updating the migration page available under `docs/migration/v11.x-carbon-components.md` to include the following migration information:

- [ ] Mapping filepaths from `carbon-components` to `@carbon/styles`
  - For example, `src/globals/scss/_colors.scss` is now at `scss/_colors.scss`
- [ ] Map common usage scenarios with `carbon-components` over to `@carbon/styles`
  - For example:
    - Here's what's changed in order to change the prefix
    - Here's what's changed in order to change the theme
    - In other words, here's the new way to do ___
  - Scenarios
    - Changing prefix
    - Specifying the theme
    - Passing in feature flags
    - Bringing in specific components
    - Utilities
    - Using stuff from vendor
- [ ] Document things in `carbon-components` that may still need to be implemented/migrated in `@carbon/styles`

_Note: some of these will overlap with usage docs from: https://github.com/carbon-design-system/carbon/issues/9169_

This will not include the following, as it is covered elsewhere:

- Design Token updates

### PR DESCRIPTION
**Pull request by joshblack**
Closes #9171 

This PR adds in the initial migration document for `carbon-components`. There are several TODO items that will need to be addressed in a future sprint that are called out in the document.

#### Changelog

**New**

**Changed**

- Update v11.x document with more blurbs
- Update `carbon-components` with all "public API files" and their changes included

**Removed**

#### Testing / Reviewing

- Take a look at the structure of the migration doc and see if it makes sense
- Review several of the blurbs for migration of a specific file and see if the format makes sense
- Leave any comments, questions, or concerns that you might have! I'd love to hear them

### CODE REVIEW

File: docs/migration/11.x-carbon-components.md [hunk 1]
```diff
@@ -2,73 +2,635 @@
...
+- `scss/components/loading/mixins`;
+- `scss/components/notification/mixins`;
+- `scss/components/tag/mixins`;
+- `scss/components/structured-list/mixins`;
+
+In v11, this entrypoint has been removed. If you are using a mixin from this
+entrypoint, find the appropriate mixin from the migration guidance for one of
+the files above.
+
+TODO: anything else we can offer for guidance here?
```
Comment (andreancardona) [inline docs/migration/11.x-carbon-components.md:239]: I think this is good for now :)

File: docs/migration/11.x-carbon-components.md [hunk 2]
```diff
@@ -2,73 +2,635 @@
...
+
+```scss
+@use '@carbon/styles/scss/components/accordion';
+```
+
+## Configuration
+
+In v10, you could configure various aspects of Carbon using global flags that
+were set before including Carbon. For example, you could change the prefix used
+for.examples or disable the CSS reset that the project used by doing the
```
Comment (andreancardona) [inline docs/migration/11.x-carbon-components.md]:
```suggestion
or disable the CSS reset that the project used by doing the
```

File: docs/migration/v11.x.md [hunk 3]
```diff
@@ -40,68 +41,56 @@ publishing.
...
-`$feature-flags` before importing `carbon-components`. For example:
+Starting in v11, we will prefix many of the packages we ship with the `@carbon`
+scope. We will also try and consolidate packages where possible to get as close
+to a one-package, one-install experience.
 
-```scss
-$feature-flags: (
-  enable-v11-release: true,
-);
+If you're using React today, you need to install several packages:
```
Comment (andreancardona) [inline docs/migration/v11.x.md]:
```suggestion

```

Maybe removing this sentence and 3 packages

I felt like this little blurb was a bit confusing as a reader, and instead going straight line 58. `In v11....etc.`

**Comment by andreancardona** [comment]
This is awesome by the way! @joshblack

**Comment by joshblack** [comment]
Let me know if you have any questions @tw15egan @jnm2377!

**Review by andreancardona — APPROVED** [review]

**Review by sstrubberg — APPROVED** [review]
So much docs!

**Review by tw15egan — APPROVED** [review]
This looks really awesome, nice work @joshblack 💯

**Pull request merged**